core/src/main/scala-3/cats/derived/DerivedOrder.scala

```diff
@@ -0,0 +1,45 @@
+package cats.derived
+
+import cats.{Order, Show}
+
+import scala.compiletime.*
+import shapeless3.deriving.{Complete, Continue, K0, Labelling}
+
+import scala.annotation.targetName
+import scala.deriving.Mirror
+
+type DerivedOrder[A] = Derived[Order[A]]
+object DerivedOrder:
+  type Or[A] = Derived.Or[Order[A]]
+
+  inline def apply[A]: Order[A] =
+    import DerivedOrder.given
+    summonInline[DerivedOrder[A]].instance
+
+  @targetName("product")
+  given [A](using inst: => K0.ProductInstances[Or, A]): DerivedOrder[A] =
```

Comment:
Let's just use the name
```suggestion
  given product[A](using inst: => K0.ProductInstances[Or, A]): DerivedOrder[A] =
```